firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -837,6 +837,16 @@ def pbs_for_set(document_path, document_data, option):
     return write_pbs
 
 
+def _convert_simple_field_paths_with_leading_digits(field_paths):
+    new_field_paths = []
+    for field_path in field_paths:
+        if field_path[0] in '0123456789':
+            new_field_paths.append("`{}`".format(field_path))
```

Comment:
This doesn't make any sense to me. Why is the fix to detect leading integers and then wrap the string with "``"? Shouldn't we use ensure the keys are all unicode?